execution/executor/src/chunk_executor.rs

```diff
@@ -0,0 +1,293 @@
+                .local_synced_version(latest_view.version().unwrap_or(0))
+                .first_version_in_request(first_version_in_request)
+                .num_txns_in_request(num_txns),
+            "sync_request_executed",
+        );
+
+        Ok(())
+    }
+
+    fn apply_chunk(
+        &self,
+        txn_output_list_with_proof: TransactionOutputListWithProof,
+        verified_target_li: &LedgerInfoWithSignatures,
+        epoch_change_li: Option<&LedgerInfoWithSignatures>,
+    ) -> Result<()> {
+        let _timer = DIEM_EXECUTOR_APPLY_CHUNK_SECONDS.start_timer();
+
+        let num_txns = txn_output_list_with_proof.transactions_and_outputs.len();
+        let first_version_in_request = txn_output_list_with_proof.first_transaction_output_version;
+        let (persisted_view, latest_view) = self.commit_queue.lock().persisted_and_latest_view();
+
+        // Verify input transaction list.
+        txn_output_list_with_proof
+            .verify(verified_target_li.ledger_info(), first_version_in_request)?;
+
+        // Skip transactions already in ledger.
+        let txns_to_skip = txn_output_list_with_proof.proof.verify_extends_ledger(
+            latest_view.txn_accumulator().num_leaves(),
+            latest_view.txn_accumulator().root_hash(),
+            first_version_in_request,
+        )?;
+        let mut txns_and_outputs = txn_output_list_with_proof.transactions_and_outputs;
+        txns_and_outputs.drain(..txns_to_skip as usize);
+
+        // Apply transaction outputs.
+        let state_view = self.state_view(&latest_view, &persisted_view);
+        let chunk_output = ChunkOutput::by_transaction_output(txns_and_outputs, state_view)?;
+        let executed_chunk = Self::apply_chunk_output_for_state_sync(
+            verified_target_li,
+            epoch_change_li,
+            &latest_view,
+            chunk_output,
+            &txn_output_list_with_proof.proof.transaction_infos[txns_to_skip..],
+        )?;
+
+        // Add result to commit queue.
+        self.commit_queue.lock().enqueue(executed_chunk);
+
+        info!(
+            LogSchema::new(LogEntry::ChunkExecutor)
+                .local_synced_version(latest_view.version().unwrap_or(0))
+                .first_version_in_request(first_version_in_request)
+                .num_txns_in_request(num_txns),
+            "sync_request_applied",
+        );
+
+        Ok(())
+    }
+
+    fn commit_chunk(&self) -> Result<Vec<ContractEvent>> {
+        let _timer = DIEM_EXECUTOR_COMMIT_CHUNK_SECONDS.start_timer();
+
+        Ok(self.commit_chunk_impl()?.all_events())
+    }
+
+    fn execute_and_commit_chunk(
+        &self,
+        txn_list_with_proof: TransactionListWithProof,
+        verified_target_li: &LedgerInfoWithSignatures,
+        epoch_change_li: Option<&LedgerInfoWithSignatures>,
+    ) -> Result<Vec<ContractEvent>> {
+        // Re-sync with DB, make sure the queue is empty.
+        self.reset()?;
```

Comment:
It can be that we gave up what's left in the queue? these two x_and_commit_chunk() interfaces will be in tests only anyway :P

Reply:
I'd think if we wanted to give up on the queue, we'd call `reset()` ourselves? Likewise, if it's test-only we could use test-only complication But yeah, don't worry too much about it. It's nitty.

Reply:
I'd just recreate the executor in case it's given up.

Yeah, it's now used by state sync v1. Once you move away from it, the compiler will complain it's not being used, I think?